### DISCUSSION TITLE
entropy calculating by entropy(i)

### DISCUSSION THREAD

**Discussion by parvinteimouriphysics** [jcmgray/quimb#257] [General]
hello
I have calculated a ground state of a finite site Heisenberg chain using MRG. Now
 I want to calculate the Von Nueman entropy of it according the site numbers,
In other words, at each step, I divide the chain(MPS)  into two parts, left and right, and move one site to the right
I add the right side of the system and calculate the entropy of the left system. I use the  ' entropy(i) 'function for this, but it's strange that I get answers greater than one! Where could the problem be?

**Comment by jcmgray** [comment]
Hi @parvinteimouriphysics, in general the entropy can be linear in the size of the subsystem (in this case whichever size is smaller out of left and right blocks). If calculated in base 2 the maximum for a system of qubits / spin-halves is exactly the size of the subsystem (a factor smaller if using natural logarithm).

For an MPS, the entropy between a bi-partition is also bounded by $\log(\chi)$ (where $\chi$ is the bond dimension), which can in general be greater than 1 also.

I didn't fully understand your description, but were you looking to calculate the entropy of site `i` only? Or the entropy of a bipartition?

**Reply by parvinteimouriphysics** [reply]
Thank you for your explanation.
I have an initial MPS and at each step I divide the chain into two parts and calculate the entropy of the system when the system is divided into two parts from site A. The purpose of this work is to calculate the central charge of the system (according to the following formula where L is the length of the whole chain and x_L is the site from where the system is divided into two parts):
$𝑆(𝑥_{𝐿})=\frac{C(1+\eta_{𝐵𝐶})}{6} \log(\frac{𝐿}{\pi}\sin(\frac{𝜋𝑥_𝐿}{𝐿})) $
$ \eta_{BC} $ is zero for OBC and 1 for PBC and C is 'central charge'
I got this plot:
<img width="620" alt="Screenshot 2024-08-28 at 9 50 46 AM" src="https://github.com/user-attachments/assets/b254160a-ea26-4358-883c-5a9d25bb837f">
and I want to find C by fitting this plot and above formula
